### PR TITLE
Hotfix/v1.0.2: Hotfix in Development

### PR DESCRIPTION
**Pull request by waji46**
**Title: Hotfix in Development**

    [P.S]
      - Updated the Readme File
      - Updated the init_delete.sh 
      - Updated the init.sh file
      - Added the resources for KMS
      - Fixed the domain issue

### CODE REVIEW

**Comment by faheem-khau9** [comment]
:robot: OpenAI








---

### Chat with 🤖 OpenAI Bot (`@openai`)
- Reply on review comments left by this bot to ask follow-up questions. A review comment is a comment on a diff or a file.
- Invite the bot into a review comment chain by tagging `@openai` in a reply.

### Code suggestions
- The bot may make code suggestions, but please review them carefully before committing since the line number ranges may be misaligned. 
- You can edit the comment made by the bot and manually tweak the suggestion if it is slightly off.

### Ignoring further reviews
- Type `@openai: ignore` anywhere in the PR description to ignore further reviews from the bot.

---






<details>
<summary>Files not summarized due to errors (15)</summary>

### Failed to summarize

* CHANGELOG.md (nothing obtained from openai)
* ROADMAP.md (nothing obtained from openai)
* infrastructure/init.sh (nothing obtained from openai)
* README.md (nothing obtained from openai)
* infrastructure/modules/serverless/cloudtrail.tf (nothing obtained from openai)
* infrastructure/modules/serverless/data.tf (nothing obtained from openai)
* infrastructure/init_delete.sh (nothing obtained from openai)
* infrastructure/main.tf (nothing obtained from openai)
* infrastructure/modules/serverless/variables.tf (nothing obtained from openai)
* infrastructure/modules/xc3/data.tf (nothing obtained from openai)
* infrastructure/modules/xc3/loadbalancer.tf (nothing obtained from openai)
* infrastructure/modules/xc3/startup-script.sh.tpl (nothing obtained from openai)
* infrastructure/terraform.auto.tfvars (nothing obtained from openai)
* infrastructure/modules/xc3/variables.tf (nothing obtained from openai)
* infrastructure/variables.tf (nothing obtained from openai)

</details>


---
In the recent run, only the files that changed from the `base` of the PR and between `cfb275065175d9813291fb636c958118f7a80ed7` and `5de1973b62b3a9856257c8a50e10798777eae26f` commits were reviewed.

<details>
<summary>Files not reviewed due to errors in the recent run (15)</summary>

### Failed to review in the last run

* ROADMAP.md (no response)
* README.md (no response)
* infrastructure/init.sh (no response)
* CHANGELOG.md (no response)
* infrastructure/modules/serverless/data.tf (no response)
* infrastructure/main.tf (no response)
* infrastructure/modules/serverless/cloudtrail.tf (no response)
* infrastructure/init_delete.sh (no response)
* infrastructure/modules/xc3/loadbalancer.tf (no response)
* infrastructure/modules/xc3/data.tf (no response)
* infrastructure/modules/serverless/variables.tf (no response)
* infrastructure/modules/xc3/startup-script.sh.tpl (no response)
* infrastructure/variables.tf (no response)
* infrastructure/modules/xc3/variables.tf (no response)
* infrastructure/terraform.auto.tfvars (no response)

</details>

File: infrastructure/init_delete.sh [hunk 1]
```diff
@@ -38,18 +39,25 @@ fi
 
 # Input variable KMS key ID
 key_id=$1
```
Comment (samanXgrid) [inline infrastructure/init_delete.sh:41]: this variable needs to be removed

File: infrastructure/init_delete.sh [hunk 2]
```diff
@@ -89,7 +97,7 @@ else
     echo "Failed to delete kms key alias"
 fi
 # Delete KMS key
-if aws kms schedule-key-deletion --key-id ${key_id} --pending-window-in-days 7 --region "${aws_region}"; then
+if aws kms schedule-key-deletion --key-id "${key_id}" --pending-window-in-days 7 --region "${aws_region}"; then
```
Comment (samanXgrid) [inline infrastructure/init_delete.sh:100]: we need to remove it from init.sh it is conflicting in 2 PRs

**Review by samanXgrid — COMMENTED** [review]
left few comments KMS key change is in two PRs so it will cause conflicts

File: infrastructure/init_delete.sh (hunk 1 above)
Comment (waji46) [reply infrastructure/init_delete.sh]: Removed this variable key_id=$1

File: infrastructure/init_delete.sh (hunk 2 above)
Comment (waji46) [reply infrastructure/init_delete.sh]: Removed these commands from the init_delete.sh file

File: infrastructure/modules/serverless/cloudtrail.tf [hunk 3]
```diff
@@ -15,16 +15,18 @@
...
   tags   = merge(local.tags, tomap({ "Name" = "${var.namespace}-bucket" }))
 
 }
 
-resource "aws_s3_bucket_acl" "this" {
-  bucket = aws_s3_bucket.this.id
-  acl    = "private"
-  #   depends_on = [aws_s3_bucket_ownership_controls.s3_bucket_acl_ownership]
-}
+# resource "aws_s3_bucket_acl" "this" {
```
Comment (muhdsaad) [inline infrastructure/modules/serverless/cloudtrail.tf:25]: Why are we commenting this code, do we need it or just commenting it out. If we don't need this you can remove it

File: infrastructure/modules/serverless/cloudtrail.tf [hunk 4]
```diff
@@ -15,16 +15,18 @@
 # Create an S3 bucket for storing CloudTrail logs
```
Comment (muhdsaad) [inline infrastructure/modules/serverless/cloudtrail.tf:15]: run terraform fmt

File: infrastructure/modules/serverless/data.tf [hunk 5]
```diff
@@ -12,6 +12,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-data "aws_kms_key" "this" {
-  key_id = "alias/${var.namespace}-kms-key"
-}
+# data "aws_kms_key" "this" {
```
Comment (muhdsaad) [inline infrastructure/modules/serverless/data.tf:15]: Do we need this or not? If not remove it

File: infrastructure/variables.tf [hunk 6]
```diff
@@ -142,3 +137,8 @@ variable "slack_channel_url" {
   type        = string
   default     = ""
 }
+
+variable "create_kms" {
+  description = "Fetch the KMS if exist"
```
Comment (muhdsaad) [inline infrastructure/variables.tf:142]: Create or Fetch?

**Review by muhdsaad — CHANGES_REQUESTED** [review]
Minor change is required

File: infrastructure/variables.tf (hunk 6 above)
Comment (waji46) [reply infrastructure/variables.tf:142]: Create

File: infrastructure/modules/serverless/cloudtrail.tf (hunk 3 above)
Comment (waji46) [reply infrastructure/modules/serverless/cloudtrail.tf:25]: Removed

File: infrastructure/variables.tf (hunk 6 above)
Comment (muhdsaad) [inline infrastructure/variables.tf:142]:
```suggestion
  description = "Check if the user wants to create KMS or not"
```

**Review by muhdsaad — CHANGES_REQUESTED** [review]

**Review by samanXgrid — APPROVED** [review]
LGTM

File: infrastructure/modules/serverless/cloudtrail.tf (hunk 4 above)
Comment (waji46) [reply infrastructure/modules/serverless/cloudtrail.tf:15]: Done

File: infrastructure/modules/serverless/data.tf (hunk 5 above)
Comment (waji46) [reply infrastructure/modules/serverless/data.tf]: Removed

File: CHANGELOG.md [hunk 7]
```diff
@@ -5,6 +5,21 @@ All notable changes to the `XgridInc/xc3` repository will be documented in this
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+
+## [1.0.1]  - 2023-06-06
+
+
+### Changed
+- Updated the startup script of XC3's EC2 instance
```
Comment (faheem-khau9) [inline CHANGELOG.md:13]: Should be Fixed

File: README.md [hunk 8]
```diff
@@ -79,30 +79,31 @@ Check the below video for a quick demo of XC3.
...
     git clone https://github.com/XgridInc/xc3.git
     ```
-2. Install the `prometheus client` library using following commands:
-    ```bash
+2. You have to configure the config.sh file in xc3/infrastructure and run the below command
+    ```
+    cd xc3/infrastructure
+    bash init.sh
+    ```
+3. Install the `prometheus-client` library using following commands:
```
Comment (faheem-khau9) [inline README.md:94]:
```suggestion
3. Zip the `prometheus-client` library using following commands:
```

File: infrastructure/modules/serverless/data.tf [hunk 9]
```diff
@@ -12,6 +12,3 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
```
Comment (faheem-khau9) [inline infrastructure/modules/serverless/data.tf:14]: Remove this file, if we don't have any code in this.

File: infrastructure/terraform.auto.tfvars [hunk 10]
```diff
@@ -15,15 +15,15 @@
...
+  "eu-west-1a" = "10.0.0.0/24"
+  "eu-west-1c" = "10.0.1.0/24"
 }
-domain_name                = "example.co"
+domain_name                = ""
 hosted_zone_id             = "Z053166920YP1STI0EK5X"
-private_subnet_cidr_block  = "11.1.1.11/26"
-allow_traffic              = ["11.11.111.111/32"]
+private_subnet_cidr_block  = "10.0.100.0/24"
+allow_traffic              = ["11.11.111.111/32"] // Put your own ip address
```
Comment (faheem-khau9) [inline infrastructure/terraform.auto.tfvars:26]:
```suggestion
allow_traffic              = ["0.0.0.0/0"] // Use your own network CIDR
```

File: infrastructure/variables.tf (hunk 6 above)
Comment (waji46) [reply infrastructure/variables.tf:142]: Done

**Review by faheem-khau9 — CHANGES_REQUESTED** [review]
Minor changes are required.

File: CHANGELOG.md (hunk 7 above)
Comment (waji46) [reply CHANGELOG.md]: Done

File: infrastructure/modules/serverless/data.tf (hunk 9 above)
Comment (waji46) [reply infrastructure/modules/serverless/data.tf]: Removed

File: infrastructure/terraform.auto.tfvars (hunk 10 above)
Comment (waji46) [reply infrastructure/terraform.auto.tfvars]: Addressed the feedback.

**Review by faheem-khau9 — DISMISSED** [review]
LGTM

**Review by yaseenzafar123 — DISMISSED** [review]

**Review by faheem-khau9 — APPROVED** [review]
LGTM

**Review by muhdsaad — APPROVED** [review]

**Review by yaseenzafar123 — APPROVED** [review]
LGTM

**Pull request merged**